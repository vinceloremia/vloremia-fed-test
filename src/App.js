import React from 'react';
import { useObserver } from 'mobx-react';
import storeInstance from './store/UserStore';

export default function App() {
	const [loading, setLoading] = React.useState('...');
	React.useEffect(() => {
		storeInstance.fetchUsers().then(() => setLoading());
	}, []);
	return useObserver(() => (
		<>
			{loading}
			{storeInstance.users.map(user => (
				<div className="card" key={user.cell}>
					<div className="imageContainer">
						<img src={user.picture.large} alt="Avatar" />
					</div>
					<div className="container">
						<h4>
							{user.name.title} {user.name.first} {user.name.last}
						</h4>
						<p>{user.email}</p>
						<p>{user.cell}</p>
					</div>
				</div>
			))}
		</>
	));
}
