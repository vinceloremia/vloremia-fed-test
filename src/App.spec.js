import React from 'react';
import App from './App';

describe('App', () => {
	it('renders the Hello world', () => {
		const actual = shallow(<App />);
		expect(actual.get(0)).toMatchSnapshot();
	});
});
