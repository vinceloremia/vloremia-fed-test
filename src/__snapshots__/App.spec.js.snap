// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the Hello world 1`] = `
<div>
  Hello world
</div>
`;
